### PR TITLE
LIME-453:  Error boundaries for filter count

### LINKED ISSUE #453: BUG: Unexpected application error when logged in user tries to reach site
### Describe your issue

Unexpected application error when logged in user tries to reach site

### Steps to reproduce

1. Sign in with valid credentials
2. Pay attention to the page displayed
3. Enter URL of any other page, different from Overview
4. Pay attention to the page displayed

### What was the actual result?

Corresponding page is displayed for a moment, then error message is displayed to the user

### What was the expected result?

Content of corresponding page is displayed to the user

### Put here any screenshots or videos (optional)

image

### What OS are you seeing the problem on?

Windows

### What browsers are you seeing the problem on?

Chrome

### Severity

Blocker (S1)

### PR DESCRIPTION
**Pull request by DavidCharif**
App is breaking
image
image
Solution:

image

### CODE REVIEW

**Comment by valery-chumak** [comment]
You are receiving notifications data as an object
image
So the error is appearing because object does not have filter function. You should do this on items array

**Comment by nikita-remeslov** [comment]
Why add so many changes here? Looks like we just need to fix notification mapping

**Comment by DavidCharif** [comment]
@nikita-remeslov  > Why add so many changes here? Looks like we just need to fix notification mapping

I didn't test it using a local database, so I got new errors on console, like a missing key, a button with a button inside, and a content type error on delete

the main issue has been solved

File: frontend/src/bundles/notifications/notifications-api.ts [hunk 1]
```diff
@@ -20,7 +20,9 @@ class NotificationsApi extends BaseHttpApi {
         super({ path: ApiPath.NOTIFICATIONS, baseUrl, http, storage });
     }
 
-    public async fetchNotifications(): Promise<Array<NotificationResponseDto>> {
+    public async fetchNotifications(): Promise<{
+        items: NotificationResponseDto[];
+    }> {
```
Comment (dimapopovych) [inline frontend/src/bundles/notifications/notifications-api.ts]:
```suggestion
    public async fetchNotifications(): Promise<{
        items: Paged<NotificationResponseDto>;
    }> {
```
Response shape still seems to be wrong, double-check, but I believe that shape should be like this.

**Review by dimapopovych — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/notifications/notifications-api.ts (hunk 1 above)
Comment (DavidCharif) [reply frontend/src/bundles/notifications/notifications-api.ts]: Looks cleaner! https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/pull/454/commits/c66bf7077220ec349e1aa89e915e8000e38c297b

**Comment by DavidCharif** [comment]
https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/53876827/561a5bd6-870e-4f6a-8492-049970a54ca6

**Comment by DavidCharif** [comment]
Added error and loading handling

https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/53876827/35a19af3-8c4b-4f0a-b28a-b2fa60fbb03d

**Comment by nikita-remeslov** [comment]
Please check this PR - https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/pull/458

**Pull request closed without merging**